### ISSUE TITLE
bug(parser): false postive for "Missing initializer in const declaration"

### ISSUE THREAD

**Issue by Boshen** [oxc-project/oxc#1723]
In DefinitelyTyped:

```
  × Missing initializer in const declaration
     ╭─[types/node/crypto.d.ts:162:1]
 162 │         /** Specifies the active default cipher list used by the current Node.js process  (colon-separated values). */
 163 │         const defaultCipherList: string;
     ·               ─────────────────
 164 │     }
     ╰────
```

1. It should not generate the error for `.d.ts`.
2. It should not parse and lint `.d.ts` files (?)
